### PR TITLE
add secure cookie flag

### PR DESCRIPTION
**Pull request by akashrpo**
## Description of the change

> Support for secure cookie flag

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

---
This change is [Reviewable](https://reviewable.io/reviews/rudderlabs/rudder-sdk-js/398)

### CODE REVIEW

File: analytics.js [hunk 1]
```diff
@@ -932,6 +935,10 @@ class Analytics {
     if (options && options.setCookieDomain) {
       this.storage.options({ domain: options.setCookieDomain });
     }
+    if (options && options.secureCookie) {
+      this.secureCookie = options.secureCookie;
+    }
+    this.storage.options({ secure: this.secureCookie });
```
[the remark this answers is not in the document]
Comment (akashrpo) [reply analytics.js]: By default we are setting the secure flag to `true` so if we do not pass any options in load() function then the cookie will be secure.

[the remark this answers is not in the document]
Comment (saikumarrs) [reply analytics.js]: It should not be defaulted to true. In the previous SDK implementation, secure flags were not added to cookies.

[the remark this answers is not in the document]
Comment (akashrpo) [reply analytics.js]: updated to set them at the end

[the remark this answers is not in the document]
Comment (akashrpo) [reply analytics.js]: currently it will only be set once `secureCookie` option is provided during load()

**Pull request merged**